offline-example/batch_inf_offline_dist.py

```diff
@@ -0,0 +1,106 @@
+## Adopted from the example provided in the vllm documentation
+
+import argparse
+from typing import Any, Dict, List
+
+import numpy as np
+import ray
+from packaging.version import Version
+from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy
+from vllm import LLM, SamplingParams
+
+# Check Ray version
+assert Version(ray.__version__) >= Version(
+    "2.22.0"), "Ray version must be at least 2.22.0"
+
+
+# Parse input arguments
+parser = argparse.ArgumentParser(description="Run batch inference with Ray Data.")
+parser.add_argument('--prompt_file', type=str, required=True, help="Path to the prompt text file.")
+parser.add_argument('--tensor_parallel_size', type=int, default=1, help="Tensor parallel size per instance.")
+parser.add_argument('--model_location', type=str, required=True, help="Model location for the LLM.")
+args = parser.parse_args()
+
+# Set tensor parallelism and prompt file from arguments
+tensor_parallel_size = args.tensor_parallel_size
+prompt_file = args.prompt_file
+model_location = args.model_location
+
+# Set sampling parameters
+sampling_params = SamplingParams(temperature=0.8, top_p=0.95)
+num_instances = 1  # Set the number of instances. Each instance will use tensor_parallel_size GPUs.
+
+
+# Create a class to do batch inference.
+class LLMPredictor:
+    def __init__(self):
+        # Create an LLM with specified tensor parallel size.
+        self.llm = LLM(model=model_location, tensor_parallel_size=tensor_parallel_size)
+
+    def __call__(self, batch: Dict[str, np.ndarray]) -> Dict[str, list]:
+        # Generate texts from the prompts.
+        outputs = self.llm.generate(batch["text"], sampling_params)
+        prompt: List[str] = []
+        generated_text: List[str] = []
+        for output in outputs:
+            prompt.append(output.prompt)
+            generated_text.append(' '.join([o.text for o in output.outputs]))
+        return {
+            "prompt": prompt,
+            "generated_text": generated_text,
+        }
+
+
+# Load the prompt file into Ray Data
+ds = ray.data.read_text(prompt_file)
+
+
+# Define scheduling strategy for tensor parallelism
+def scheduling_strategy_fn():
```

Comment:
What does this scheduling strategy function do?

Reply:
It uses the ray's placement group which  creates resources for data processing. In the  70B example, this would create a resource group of 4GPUs for scheduling the tasks. More info in here: https://docs.ray.io/en/latest/ray-core/api/doc/ray.util.placement_group.html